src/main/java/com/greenfox/tribes/services/ShopService.java

```diff
@@ -13,23 +15,34 @@
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.List;
 
 @Service
 @AllArgsConstructor
 public class ShopService {
 
   private EquipmentRepository equipmentRepository;
+  private PersonaService personaService;
   private EquipmentService equipmentService;
   private UserRepository userRepository;
   private CharacterEquipmentRepository characterEquipmentRepository;
 
   public ArrayList<ShopItemDTO> getShoppingList() {
     Authentication auth = SecurityContextHolder.getContext().getAuthentication();
     ArrayList<ShopItemDTO> shoppingList = new ArrayList<>();
-    for (Equipment equipment : equipmentRepository.findAll()) {
-      int numberOwned =
-          characterEquipmentRepository.countAllByEquipmentAndPersona(
-              equipment, userRepository.findByUsername(auth.getName()).get().getPersona());
+    PersonaDTO personaDTO = personaService.readCharacter();
+    List<EquipmentDTO> backpackItems = personaService.showEquipment(personaDTO);
+    List<Equipment> allEquipment = equipmentRepository.findAll();
+
+    for (Equipment equipment : allEquipment) {
+      int numberOwned = 0;
+      for (EquipmentDTO e : backpackItems) {
+        CharacterEquipment characterEquipment =
+            characterEquipmentRepository.findById(e.getId()).orElse(null);
```

Comment:
I think this is there from the previous branch. Don't do this, you could throw an exception that will return 4xx error instead of 5xx when you get null pointer exception somewhere